vaadin-renderer-flow-parent/vaadin-renderer-flow/src/main/java/com/vaadin/flow/renderer/LitRenderer.java

```diff
@@ -62,25 +64,23 @@
  *      "https://cdn.vaadin.com/vaadin-web-components/20.0.0/#/elements/vaadin-combo-box"><code>&lt;vaadin-combo-box&gt;.renderer</code></a>
  */
 @JsModule("./lit-renderer.ts")
-public class LitRenderer<T> extends Renderer<T> {
+public class LitRenderer<SOURCE> implements Serializable {
+    private final static AtomicInteger instanceCounter = new AtomicInteger(0);
     private final String templateExpression;
 
     private final String DEFAULT_RENDERER_NAME = "renderer";
 
     private final String propertyNamespace;
 
-    private final Map<String, ValueProvider<T, ?>> valueProviders = new HashMap<>();
-    private final Map<String, SerializableBiConsumer<T, JsonArray>> clientCallables = new HashMap<>();
+    private final Map<String, ValueProvider<SOURCE, ?>> valueProviders = new HashMap<>();
+    private final Map<String, SerializableBiConsumer<SOURCE, JsonArray>> clientCallables = new HashMap<>();
 
-    private LitRenderer(String templateExpression) {
+    protected LitRenderer(String templateExpression) {
         this.templateExpression = templateExpression;
 
-        // Generate a unique (in scope of the UI) namespace for the renderer
-        // properties.
-        int litRendererCount = UI.getCurrent().getElement()
-                .getProperty("__litRendererCount", 0);
-        UI.getCurrent().getElement().setProperty("__litRendererCount",
-                litRendererCount + 1);
+        // Generate a unique (in scope of the JVM instance) namespace for
+        // the renderer properties.
+        int litRendererCount = instanceCounter.incrementAndGet();
```

Comment:
Using a static counter for the namespace may be problematic since the accumulated number (available for everyone) could be used to get an approximation of how many users an application has.